Celeste.Mod.mm/Mod/Registry/StrawberryRegistry.cs

```diff
@@ -0,0 +1,154 @@
+        public static void Register(Type type, bool tracked = true, bool alternateCollectionRules = false)
+        {
+            if (Attribute.IsDefined(type, typeof(CustomEntityAttribute)))
+            {
+                CustomEntityAttribute attr = Attribute.GetCustomAttribute(type, typeof(CustomEntityAttribute)) as CustomEntityAttribute;
+                foreach (string id in attr.IDs)
+                {
+                    registeredBerries.Add(new RegisteredBerry(type, id, tracked, alternateCollectionRules));
+                }
+            }
+        }
+
+        public static bool TrackableContains(string name)
+        {
+            ReadOnlyCollection<RegisteredBerry> berries = GetTrackableBerries();
+            foreach (RegisteredBerry berry in berries)
+            {
+                if (berry.entityName == name)
+                    return true;
+            }
+            return false;
+        }
+
+        // Is it the first normally collectable strawberry in the train?
+        public static bool IsFirstStrawberry(Entity self)
+        {
+            Follower follow = self.Components.Get<Follower>();
+            if (follow == null)
+                return false;
+
+            // we need it searchable and Find doesn't work on a ReadOnlyCollection aaaaa
+            List<RegisteredBerry> berries = registeredBerries.ToList();
+            ReadOnlyCollection<Type> types = GetBerryTypes();
+            //RegisteredBerry detectedBerry = berries.Find(b => b.GetType() == self.GetType());
+            //if (detectedBerry != null)
+            //if (detectedBerry == null)
+            //    return false;
+
+            for (int i = follow.FollowIndex - 1; i >= 0; i--)
+            {
+                Entity strawberry = follow.Leader.Followers[i].Entity;
+                bool goldenCheck = false;
+                if (strawberry as Strawberry != null)
+                    goldenCheck = (strawberry as Strawberry).Golden;
+
+                //Strawberry vanillaStraw = strawberry as Strawberry;
+                //if (vanillaStraw != null && !vanillaStraw.Golden)
+                //{
+                //    return false;
+                //}
+                Type t = follow.Entity.GetType();
+                RegisteredBerry berry = berries.Find(b => b.berryClass == t);
+                bool blocksCollect = berry.blocksNormalCollection;
+                if (types.Contains(strawberry.GetType()) && 
+                    berry.blocksNormalCollection == false && 
+                    !goldenCheck)
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        // Called by CoreMapDataProcessor in order to inject mod berries into the step list for fixup.
+        public static Dictionary<string, Action<BinaryPacker.Element>> GetBerriesToInject(MapDataFixup context)
```

Comment:
Instead of `CoreMapDataProcessor.Reset` calling this and adding the entries to its steps dict, you could make `CoreMapDataProcessor.Run` check if the berry is trackable and call `Context.Run` instead.

Reply:
That is _much_ more reasonable and I hate that I missed that. Consider it done. ..Literally as of this moment.